tsfc/fem.py

```diff
@@ -233,6 +234,11 @@ def flat_index(ordered_deriv):
         return result[0]
 
 
+def zero_tracking_literal(table, ufl_element):
+    track_zeros = isinstance(ufl_element, (ufl.EnrichedElement, ufl.MixedElement))
```

Comment:
Correct.

One more comment: COFFEE could handle zeros in the middle too, it's just that the performance experiments I run time ago showed that vectorisation was much worse by creating jumps "right in the middle" on the iteration space. I actually had forgotten about this until I looked again at the code